File: FoodieApplication/foodInventoryService/src/main/java/com/example/foodInventoryService/model/Restaurant.java
package com.example.foodInventoryService.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Document
public class Restaurant {

    @Id
    private String restaurantId;
    private String restaurantName;
    private String restaurantLocation;
    private List<Dish> dishList;
}

File: FoodieApplication/frontend/FoodieApplication/src/app/add-Restaurant-Service/addRestModel/restaurant.ts
import { Dish } from "./dish";
import { Image } from "./image";

export class Restaurant {

    restaurantId:string;
    restaurantName:string;
    restaurantLocation:string;
    image:Image[]=[];
    dishList:Dish[]=[];

    constructor()
    {
        this.restaurantId='';
        this.restaurantName='';
        this.restaurantLocation='';
        this.dishList=[];
    }
}
